File: 系统架构/MVC&&MVP&&MVVM/iOSMVP/iOSMVP/ViewController.m
//
//  ViewController.m
//  iOSMVP
//
//  Created by KB on 2017/11/30.
//  Copyright © 2017年 HZ. All rights reserved.
//

#import "ViewController.h"

#import "BookView.h"
#import "Presenter.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

-(void)layoutSubViews{
    BookView* bookView = [[BookView alloc] init];
    Presenter* presenter = [[Presenter alloc]initWithBookView:bookView];
    [presenter getBook];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 基础-语言篇/Swift/HZThreeParty/HZThreeParty/HZThreeParty/ViewController.swift
//
//  ViewController.swift
//  HZThreeParty
//
//  Created by KB on 2018/7/3.
//  Copyright © 2018年 BQ. All rights reserved.
/*
   关于导航栏与控制器顶部遮蔽问题
 
   <1> 控制器会自动将UIScrollview及其子类的内容显示区域自动调整到导航栏以下；
     可通过以下代码设置禁止自动调整
     if #available(iOS 11.0, *) {
     self.tableView?.contentInsetAdjustmentBehavior = UIScrollViewContentInsetAdjustmentBehavior.never
     }else  {
     self.automaticallyAdjustsScrollViewInsets = false;
     }
 
 */

import UIKit

class ViewController: HZListViewController {
    

    override func viewDidLoad() {
        super.viewDidLoad()
        
        dataSource = ["HZAlamofireViewController"]
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

